backend/src/middlewares/medicationMiddlewares.js

```diff
@@ -31,3 +31,25 @@ async function deleteMedication(req, res, next) {
 }
 
 export { deleteMedication };
+
+async function postMedication(req, res, next) {
+  try {
+    const existingMedication = await Medication.findOne({
+      profile: req.body.profile,
+      nameOfMedication: req.body.nameOfMedication,
+    });
+    if (existingMedication) {
+      return new ExpressError('Medication already in your aidkit', StatusCodes.CONFLICT);
+    } else {
+      const medication = new Medication({
+        ...req.body,
+      });
+      const addedMedication = await medication.save();
+      return addedMedication;
```

Comment:
We should send a response with added medication instead of returning this document